cqf-fhir-cr-hapi/src/main/java/org/opencds/cqf/fhir/cr/hapi/cdshooks/CdsCrService.java

```diff
@@ -1,333 +1,62 @@
-            }
-        }
-        resourceMap.forEach(
-                (key, value) -> BundleHelper.addEntry(prefetchResources, BundleHelper.newEntryWithResource(value)));
-        return prefetchResources;
+    public FhirVersionEnum getFhirVersion() {
+        return repository.fhirContext().getVersion().getVersion();
     }
 
-    @SuppressWarnings("unchecked")
-    public CdsServiceResponseJson encodeResponse(Object response) {
-        if (!(response instanceof IBaseResource)) {
-            throw new InternalErrorException("response is not an instance of a Resource");
-        }
-        serviceResponse = new CdsServiceResponseJson();
-        IResourceAdapter mainRequest = null;
-        IPrimitiveType<String> canonical = null;
-        if (response instanceof IBaseParameters parameters) {
-            var parametersAdapter = adapterFactory.createParameters(parameters);
-            var bundle = parametersAdapter.getParameter().stream()
-                    .map(p -> (IBaseBundle) p.getResource())
-                    .findFirst()
-                    .orElse(null);
-            if (bundle == null) {
-                throw new InternalErrorException("response does not contain a Bundle");
-            }
-            if (!BundleHelper.getEntry(bundle).isEmpty()) {
-                mainRequest = adapterFactory.createResource(BundleHelper.getEntryResourceFirstRep(bundle));
-                canonical = (IPrimitiveType<String>) mainRequest.getProperty("instantiatesCanonical")[0];
-            }
-        } else {
-            responseAdapter = adapterFactory.createResource((IBaseResource) response);
-            var activity = responseAdapter.getProperty("activity");
-            if (activity != null && activity.length > 0) {
-                var requestGroupRef = responseAdapter.resolvePath(activity[0], "reference", IBaseReference.class);
-                mainRequest = adapterFactory.createResource(resolveResource(requestGroupRef));
-                var definition = mainRequest.getProperty("definition");
-                if (definition != null && definition.length > 0) {
-                    canonical = mainRequest.resolvePath(definition[0], "reference", IPrimitiveType.class);
-                }
-            }
-        }
-
-        if (mainRequest == null || canonical == null) {
-            throw new InternalErrorException("unable to resolve response");
-        }
-
-        var planDef = (IPlanDefinitionAdapter)
-                adapterFactory.createResource(SearchHelper.searchRepositoryByCanonical(repository, canonical));
-        var links = resolvePlanLinks(planDef);
-        Stream.of(mainRequest.getProperty("action"))
-                .map(b -> adapterFactory.createRequestAction((IBaseBackboneElement) b))
-                .forEach(action -> serviceResponse.addCard(resolveAction(action, links)));
-
-        return serviceResponse;
+    public IBaseParameters encodeParams(CdsServiceRequestJson json) {
```

Comment:
what if someone needed information contained in the requestDetails to encode the Params?

to maximise extensibility, i'm thinking that we should pass the requestDetails as a parameter to the method invocation.  what do u think?